src/cubi_tk/sodar/ingest_data.py

```diff
@@ -371,90 +265,124 @@ def check_col_index(column_index):
-            out_column_index = [
-                i
-                for i, head in enumerate(assay_header)
-                if head not in ignore_cols
-                and out_column.lower()
-                in re.sub("(Parameter Value|Comment|Characteristics)", "", head).lower()
-            ]
-            out_column_index = check_col_index(out_column_index)
-
-        return {line[in_column_index]: line[out_column_index] for line in assay_lines}
-
-    def download_webdav(self, sources):
-        download_jobs = []
-        folders = []
-        for src in sources:
-            if re.match("davs://", src):
-                download_jobs.append(TransferJob(path_remote=src, path_local=self.args.tmp))
-                tmp_folder = f"tmp_folder_{len(download_jobs)}"
-                pathlib.Path(tmp_folder).mkdir(parents=True, exist_ok=True)
+            column_index = column_index[0]
+        column_dict: dict[str,str] = {}
+        for line in sheet_lines:
+            if conservation_method_idx is not None and self.args.preset == "onk_analysis":
+                #add conservation method
+                column_dict[line[sample_name_idx]] = [line[column_index], line[conservation_method_idx]]
             else:
-                folders.append(src)
-
-        if download_jobs:
-            logger.info("Planning to download folders...")
-            for job in download_jobs:
-                logger.info("{} => {}", job.path_remote, job.path_local)
-            if not self.args.yes and not input("Is this OK? [yN] ").lower().startswith("y"):
-                logger.error("OK, breaking at your request")
-                return []
-
-            counter = Value(c_ulonglong, 0)
-            total_bytes = sum([job.bytes for job in download_jobs])
-            with tqdm.tqdm(total=total_bytes) as t:
-                if self.args.num_parallel_transfers == 0:  # pragma: nocover
-                    for job in download_jobs:
-                        download_folder(job, counter, t)
-                else:
-                    pool = ThreadPool(processes=self.args.num_parallel_transfers)
-                    for job in download_jobs:
-                        pool.apply_async(download_folder, args=(job, counter, t))
-                    pool.close()
-                    pool.join()
-
-        return folders
-
-    def build_jobs(self, library_names=None) -> tuple[str, tuple[TransferJob, ...]]:
+                column_dict[line[sample_name_idx]] = line[column_index]
+        return column_dict
+
+
+    def find_collection_name(self, sample_name, column_match, m):
+        if column_match is None:
+            return sample_name
+        val = column_match[sample_name]
+        if not isinstance(val, list):
+            return val
+        #needs further matching with tumor and conservation method
+        conservation_col_name = None
+        conservation = m.groupdict(default=None)["conservation"]
+        tissue = m.groupdict(default=None)["tissue"]
```

Comment:
Maybe add a check here if these capture groups exist?
Probably catched later anyway, but might be good to raise an explicit message what went wrong?